src/ess/reduce/nexus/workflow.py

```diff
@@ -514,7 +519,10 @@ def __init__(
         self, attrs: dict[str, Any], children: dict[str, snx.Field | snx.Group]
     ):
         children = _drop(children, (snx.NXoff_geometry, snx.NXevent_data))
-        children['data'] = children['detector_number']
+        # Histogram mode detectors may not have a detector_number, as it is not needed
+        # for grouping data by detector id.
+        if 'detector_number' in children:
+            children['data'] = children['detector_number']
```

Comment:
I'm surprised the downstream code can handle this without changes. Is it possible to completely remove this line even for event data?

Reply:
You are right, but I don't know what to do then, apart from requiring a `detector_number` from ECDC...
I don't see how we can load the empty detector without loading the data to know the shape.
Loading the `x/y_pixel_offsets` wouldn't necessarily give you the shape because they can be one-dimensional (e.g. for timepix3, the `x_pixel_offset` is a 1D array of size 4096, and the same for the `y_pixel_offset`, only the `detector_number` has shape 4096x4096)

Reply:
Yes, I forgot that we can look at the shape of an hdf5 dataset without loading the data... it should be possible to create it on the fly.

Reply:
I think the overall approach is good. See my individual comments.